tests/drivers/gpio/gpio_basic_api/src/main.c

```diff
@@ -77,21 +78,33 @@ static void board_setup(void)
 static void *gpio_basic_setup(void)
 {
 	board_setup();
-
+	(void)pm_device_runtime_get(DEVICE_DT_GET(DEV));
+	if (IS_ENABLED(CONFIG_UART_CONSOLE)) {
+		(void)pm_device_runtime_get(DEVICE_DT_GET(DT_CHOSEN(zephyr_console)));
```

Comment:
This won't work if `CONFIG_UART_CONSOLE=n` and `zephyr,console` is _not_ defined in DTS, as the macro expansion will fail due to missing intermediate macros.

This `if` must be wrapped in `#if DT_HAS_CHOSEN(zephyr_console)`.